tensorrt_llm/_torch/auto_deploy/custom_ops/flashinfer_rope.py

```diff
@@ -42,7 +42,9 @@ def apply_rope_with_input_pos_flashinfer(
     q_flat = q.view(batch_size * seq_len, -1)
     k_flat = k.view(batch_size * seq_len, -1)
```

Comment:
let's not rely on batch_size or seq_len when we assume 3d/4d input... 

We can do the following:

```
position_ids = position_ids.view(-1).to(q.device)
num_nnz = position_ids.shape[0]  # this is batch_size * seq_len
q_flat = q.view(num_nnz, -1)
k_flat = k.view(num_nnz, -1)